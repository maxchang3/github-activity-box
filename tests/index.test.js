import { beforeEach, expect, it, vi } from 'vitest'
import events from './fixtures/events.json'

const mockedUpdate = vi.fn()
const mockedConsoleError = vi.fn()

vi.spyOn(console, 'log').mockImplementation(() => {})
vi.spyOn(console, 'error').mockImplementation(mockedConsoleError)
vi.spyOn(process, 'exit').mockImplementation(() => {})

vi.mock('gist-box', async (importOriginal) => {
    const actual = await importOriginal()
    return {
        ...actual,
        GistBox: vi.fn().mockImplementation(() => ({
            update: mockedUpdate,
        })),
    }
})

vi.mock('@octokit/rest', () => ({
    Octokit: vi.fn().mockImplementation(() => ({
        activity: {
            listPublicEventsForUser: vi.fn().mockResolvedValue({
                data: events,
            }),
        },
    })),
}))

describe('activity-box', () => {
    beforeEach(async () => {
        mockedUpdate.mockClear()
    })

    const runAction = async () => {
        vi.resetModules()
        return await import('../index.js')
    }

    it('updates the Gist with the expected string', async () => {
        await runAction()
        expect(mockedUpdate).toHaveBeenCalledOnce()
        expect(mockedUpdate.mock.calls[0][0]).toMatchSnapshot()
    })

    it('handles failure to update the Gist', async () => {
        mockedUpdate.mockImplementationOnce(() => {
            throw new Error('404')
        })

        await runAction()
        expect(mockedConsoleError).toHaveBeenCalledOnce()
        expect(mockedConsoleError.mock.calls[0][0]).toMatchSnapshot()
        expect(mockedUpdate).toHaveBeenCalled()
    })
})
