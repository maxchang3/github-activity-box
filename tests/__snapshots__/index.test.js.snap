// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`activity-box > handles failure to update the Gist 1`] = `"Error getting or update the Gist: Error: 404"`;

exports[`activity-box > updates the Gist with the expected string 1`] = `
{
  "content": "💪 Opened PR #4 in clippy/take-over-github
💪 Opened PR #3 in clippy/really-really-really-really-really...
🎉 Merged PR #2 in clippy/take-over-github
🗣 Commented on #1 in clippy/take-over-github
❗️ Opened issue #1 in clippy/take-over-github",
}
`;
